service/history/queueAckMgr.go

```diff
@@ -133,15 +132,21 @@ TaskFilterLoop:
 	return tasks, morePage, nil
 }
 
-func (a *queueAckMgrImpl) completeTask(taskID int64) {
+func (a *queueAckMgrImpl) completeQueueTask(taskID int64) error {
+	err := a.processor.completeTask(taskID)
```

Comment:
I'm a little confused here.  I thought we are going to cleanup the task as a separate background go-routine.  Each individual processor only updates the acklevel for the cluster.

Reply:
this will be actually called by each processor.
the replicator queue processor will actually complete the task
the transfer queue processor will do nothing